### ISSUE TITLE
Improve formatting for record field with constraint

### ISSUE THREAD

**Issue by hhugo** [ocaml-ppx/ocamlformat#374]
```
let x = { a : int = b }
let x { a : int = b } = 2
```
currently get rewritten to 
```
let x = { a = (b : int) }
let x { a = (b : int) } = 2
```

**Comment by jberdine** [comment]
Probably similar fix as #298

**Comment by gpetiot** [comment]
Seems related to a parsing issue instead: https://caml.inria.fr/mantis/view.php?id=7758
